src/client/components/share-menu/share-menu.tsx

```diff
@@ -68,20 +66,33 @@ export const ShareMenu: React.SFC<ShareMenuProps> = props => {
     onClose={onClose}
   >
     <ul className="bubble-list">
-      {<CopyToClipboard key="copy-url" text={url}>
+
+      <CopyToClipboard key="copy-url" text={getCubeViewHash(essence, true)}>
         <li onClick={onClose}>
-          {fixedTimeUrl ? STRINGS.copyRelativeTimeUrl : STRINGS.copyUrl}
+          {essence.filter.isRelative() ? STRINGS.copyRelativeTimeUrl : STRINGS.copyUrl}
+        </li>
+      </CopyToClipboard>
+
+      {essence.filter.isRelative() && <CopyToClipboard key="copy-specific-url" text={getCubeViewHash(essence.convertToSpecificFilter(timekeeper), true)}>
+        <li onClick={onClose}>
+          {STRINGS.copyFixedTimeUrl}
         </li>
       </CopyToClipboard>}
-      {fixedTimeUrl && <CopyToClipboard key="copy-specific-url" text={fixedTimeUrl}>
-        <li  onClick={onClose}>
-          {STRINGS.copyFixedTimeUrl}</li>
-      </CopyToClipboard>}
+
       {exportOptions.map(({ label, fileFormat }) =>
         <li key={`export-${fileFormat}`} onClick={() => onExport(fileFormat)}>
           {label}
         </li>
       )}
+
+      {externalViews.map((externalView: ExternalView, i: number) => {
+        const url = externalView.linkGeneratorFn(essence.dataCube, essence.timezone, essence.filter, essence.splits);
+        return <li key={`custom-url-${i}`}>
```

Comment:
Is there any unique property that can be used as key instead of array index?

Reply:
Not really. ExternalViews are taken from config and I think index is as good as any artificial ident generated from input.

Reply:
If they are static as you said, then ok.